packages/protocol/contracts/governance/VestingInstance.sol

```diff
@@ -0,0 +1,379 @@
+
+  // indicates if the withdrawing has been paused
+  bool public paused;
+
+  // the time at which the revocation has taken place
+  uint256 public revokeTime;
+
+  // the time at which the pausing shall end
+  uint256 public pauseEndTime;
+
+  // indicates if the contract is revokable
+  bool public revocable;
+
+  // revoking address and refund destination
+  address public revoker;
+  address public refundDestination;
+
+  // a public struct hosting the vesting scheme params
+  VestingScheme public vestingScheme;
+
+  event WithdrawalPaused(uint256 pausePeriod, uint256 pauseTimestamp);
+  event VestingRevoked(uint256 revokeTimestamp);
+
+  /**
+   * @notice A constructor for initialising a new instance of a Vesting Schedule contract
+   * @param vestingBeneficiary address of the beneficiary to whom vested tokens are transferred
+   * @param vestingAmount the amount that is to be vested by the contract
+   * @param vestingCliff duration in seconds of the cliff in which tokens will begin to vest
+   * @param vestingStartTime the time (as Unix time) at which point vesting starts
+   * @param vestingPeriodSec duration in seconds of the period in which the tokens will vest
+   * @param vestAmountPerPeriod the vesting amound per period where period is the vestingAmount distributed over the vestingPeriodSec
+   * @param vestingRevocable whether the vesting is revocable or not
+   * @param vestingRevoker address of the person revoking the vesting
+   * @param vestingRefundDestination address of the refund receiver after the vesting is deemed revoked
+   * @param registryAddress address of the deployed contracts registry
+   */
+  constructor(
+    address vestingBeneficiary,
+    uint256 vestingAmount,
+    uint256 vestingCliff,
+    uint256 vestingStartTime,
+    uint256 vestingPeriodSec,
+    uint256 vestAmountPerPeriod,
+    bool vestingRevocable,
+    address vestingRevoker,
+    address vestingRefundDestination,
+    address registryAddress
+  ) public {
+    // perform checks on the input data
+    require(registryAddress != address(0), "registry address must not be the zero address");
+    require(vestingAmount > 0, "Vesting amount must be positive");
+    require(vestingBeneficiary != address(0), "Beneficiary is the zero address");
+    require(vestingRefundDestination != address(0), "Refund destination is the zero address");
+    require(vestingCliff <= vestingPeriodSec, "Vesting cliff is longer than vesting duration");
+    require(vestingPeriodSec > 0, "Vesting period is 0 s.");
+    require(
+      vestAmountPerPeriod <= vestingAmount,
+      "Vesting amount per period is greater than the total vesting amount"
+    );
+    require(
+      vestingStartTime.add(vestingCliff) > block.timestamp,
+      "Vesting end time is before current time"
+    );
+
+    // set the registry address
+    setRegistry(registryAddress);
+
+    // init the vesting scheme
+    vestingScheme.vestingPeriods = vestingAmount.div(vestAmountPerPeriod);
```

Comment:
Why do we need to store this if it can be derived from `vestingAmount` and `vestAmountPerPeriod`?

Reply:
removed, gets computed when required